File: Leetcode - Java/448_array_findDisappearedNumbers.java
// https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for(int i = 0; i < nums.length; i++) {
            int idx = Math.abs(nums[i]);
            nums[idx-1] = -(Math.abs(nums[idx-1]));
        }
        List<Integer> res = new ArrayList<Integer>(); 
        for(int i = 0; i < nums.length; i++) {
            if(nums[i] > 0) {
                res.add(i+1);
            }
        }
        return res; 
    }
}

File: Leetcode - Python/448_array_findDisappearedNumbers.py
# https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/


class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        res = []

        for i in range(len(nums)):
            val = abs(nums[i]) - 1
            if nums[val] > 0:
                nums[val] = -nums[val]

        for i in range(len(nums)):
            if nums[i] > 0 :
                res.append(i + 1)

        return res